crates/cast/bin/cmd/create2.rs

```diff
@@ -132,6 +137,11 @@ impl Create2Args {
             n_threads = n_threads.min(2);
         }
 
+        let mut top_bytes = B256::ZERO;
+        if let Some(caller_address) = caller {
+            top_bytes[..20].copy_from_slice(&caller_address.into_array());
+        }
```

Comment:
This can be simplified to `let top_bytes = caller.map(|a| a.into_word()).unwrap_or_default()`

Reply:
This actually left pads the `Address` to convert to `FixedBytes<32>`:
https://docs.rs/alloy-primitives/latest/alloy_primitives/struct.Address.html#method.into_word

It would generate something like this:
```
000000000000000000000000d8da6bf26964af9d7eed9e03e53415d37aa96045
```

while we need something like this:

```
d8da6bf26964af9d7eed9e03e53415d37aa96045000000000000000000000000
```